src/api/app/models/concerns/staging_project.rb

```diff
@@ -129,6 +129,20 @@ def staging_project?
     staging_workflow_id.present?
   end
 
+  def merge_staged_requests
+    return unless overall_state == :acceptable
```

Comment:
As this changes data I would treat it like a validation and return false and an error on the instance.